File: examples/req-rep/zmq_rep.py
#! /usr/bin/env python3

"""
0MQ example: REP/REQ pattern
"""

import zmq

context = zmq.Context()
socket = context.socket(zmq.REP)

socket.bind("tcp://*:5555")

while True:
    #  Wait for next request from client
    message = socket.recv()
    print("Received request: ", message)
    socket.send(message[::-1])

File: examples/req-rep/zmq_rep.cpp
//
// zmq.hpp example: REP/REQ pattern
//

#include <zmq.hpp>
#include <string>
#include <iostream>
#include <unistd.h>

int main()
{
    //  Prepare our context and socket
    zmq::context_t context(1);
    zmq::socket_t socket(context, ZMQ_REP);
    socket.bind("tcp://*:5555");

    while (true)
    {
        zmq::message_t request;

        //  Wait for next request from client
        socket.recv(&request);
        std::string hello(static_cast<const char *>(request.data()), request.size());
        std::cout << "Received: " << hello << std::endl;

        std::reverse(hello.begin(), hello.end());

        //  Send reply back to client
        zmq::message_t reply(hello.length());
        memcpy(reply.data(), hello.data(), hello.length());
        socket.send(reply);
    }

    return 0;
}
